### ISSUE TITLE
The projects containing the static extensions are generally not part of any NuGet. 

### ISSUE THREAD

**Issue by sfilipi** [dotnet/machinelearning#2094]
I believe the projects with the name pattern Microsoft.ML.*.StaticPipe were created setting apart the static extensions so they can be on a separate nuget package. 

Most of them are currently part of no package. There exists a main package for the main project containing the static pipe: Microsoft.ML.StaticPipe. 

@TomFinley, @glebuk @shauheen Should those projects be part of that package, or separate package?

cc @abgoswam, @eerhardt 

Note: this issue can be closed when all the projects containing static extensions are part of NuGet packages, depending on the resolution of the question above.

**Comment by shauheen** [comment]
#2076 was done to address part of this issue. That NuGet should contain everything Static API.

**Comment by abgoswam** [comment]
Thanks Senja for the issue.

As part of #1914, we created separate 5 assemblies for static extensions.  These assemblies are named corresponding to nuget packages (with the Microsoft.ML prefix)

1. Microsoft.ML.StaticPipe
2. Microsoft.ML.LightGBM.StaticPipe
3. Microsoft.ML.TensorFlow.StaticPipe
4. Microsoft.ML.OnnxTransform.StaticPipe
5. Microsoft.ML.HalLearners.StaticPipe

**TODOs**

A. There should be two more assemblies that we should add to the list:
- Microsoft.ML.Recommender.StaticPipe  (based on #2077)
- Microsoft.ML.Recommender.StaticPipe (based on #1656)

B. Nuget story for the Static API ?

#2072 introduces a nuget  `Microsoft.ML.StaticPipe`.   Should we add separate nugets for the other  static assemblies as well ?

**Comment by TomFinley** [comment]
I think we should add these static pipe nugets as part of v0.10, but for v0.9 you are correct we deliberately only shipped the "main" one as part of a nuget.

The ideal situation is separate nugets, I think. It would be nice if the "core" nuget package references only a few small nugets. Anything referencing a larger nuget, should be in its own nuget and not bundled with that. That in mind @abgoswam's project should *provisionally* be considered to reflect the desired nuget story.

1. Microsoft.ML.LightGBM.StaticPipe
2. Microsoft.ML.TensorFlow.StaticPipe
3. Microsoft.ML.OnnxTransform.StaticPipe
4. Microsoft.ML.HalLearners.StaticPipe

These should each have separate nugets, if the things they depend on have separate nugets, which they all do. I considered these unessential to ship as part of v0.9 though considering the limited timeframe.

**Comment by ileve** [comment]
Until then, how can i use static extensions in namespace Microsoft.ML.TensorFlow.StaticPipe ?

**Comment by eerhardt** [comment]
@ileve - are you using a `0.10.0-preview` build from https://dotnet.myget.org/F/dotnet-core/api/v3/index.json? If so, you can reference the `Microsoft.ML.StaticPipe` package, which contains those APIs.

If not, `v0.10` will be published to nuget.org soon.

**Comment by ileve** [comment]
Thanks for the help. I installed Version 0.11.0-preview-27405-2 for the following assemblies:

```
Microsoft.ML
Microsoft.ML.StaticPipe
Microsoft.ML.TensorFlow
```

but when i try to use the extension method `Vector<float>.ApplyTensorFlowGraph` with `ctrl + .`,  no suggestions. Still missing... I also tried with version 0.10.0-preview-27330-5. Same also...

And i notice that it is not contained in the docs. But in the source code it is still there.

As I understand you will not include this in version 0.10.0 and probably not in 0.11.0?
Is there a way to access this static extension?
